app/services/make_an_offer.rb

```diff
@@ -64,9 +63,9 @@ def offer_conditions
   attr_reader :application_choice
 
   def offered_course_option
-    return nil if @course_data.nil?
+    return nil unless @course_option_id && @course_option_id != application_choice.course_option.id
```

Comment:
This seems prudent — it would be a bit untidy to have identical `offered_course_options_id`s/`course_option_id`s hanging around just because vendors had happened to send back the existing course_data in the request body